woocommerce/woocommerce-android
# Bundle UI changes

Part of: #9541

### Why
As part of the support for extensions milestone 2, we are working on support bundle products in the order creation flow.

### Description
This PR includes some UI changes to match the design team's recommendations.

> I can suggest some small improvements besides that, though:
Put selection buttons on all the rows
Disable selection buttons for items with min qty >= 1
Enable selection buttons for items with min qty == 0
Enable selection buttons for items marked as Optional
Only show the stepper when selection button is checked
If selection button is checked, set qty to 1, or min qty – whichever is higher
If qty reaches 0, deselect item and hide stepper
Disable the minus button on items that are set to their min qty
Disable the plus buttons on items that are set to their max qty

![image](https://github.com/woocommerce/woocommerce-android/assets/18119390/f5c7b7fb-b785-4c6a-a1d7-ad60f72011f5)

This PR also includes a small change to prevent updating the bundled product quantity while the product is updating.


### Testing instructions

#### Prerequisites
1. Create bundled products with different children quantity rule configurations (some of them accepting quantity 0)
2. Create a bundled product with an optional product

#### Testing
TC1
1. Create a new order with the bundled products created in prerequisites
2. On the configuration screen, check that when the optional product is selected, the stepper control is hidden
3. On the configuration screen, check that when the selected quantity of a product is 0, the stepper control is hidden

TC2
1. Open the order tap
2. Tap on create new order
3. Select a bundled product from the product selection
4. Try to update product quantity quickly using the stepper control
5. check that the stepper is disabled while the product is updating

TC3
1. Repeat the TC2 for product that is NOT a bundled product
2. Check that the stepper control keep functioning as expected

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/290db36b-bf59-401f-9f4c-7bdd85f5cef1
